Changelog.md

```diff
@@ -1,5 +1,9 @@
 ### 0.8.14 (unreleased)
 
+Important Bugfixes:
+ * ABI Encoding: When ABI-encoding values in calldata that contain nested arrays, correctly validate the nested array length against ``calldatasize()`` in all cases.
```

Comment:
Does it really affect only nested arrays? From the code it looks like it would affect 1D arrays as well. Is `calldataAccessFunction()` only called for nested arrays or something?

Reply:
``calldataAccessFunction`` is only used with the base type of an array or the member of a struct - and the bug is only in the part where that type is dynamically sized.
So
```
contract C {
  struct S { uint[] x; }
  function f(S calldata a) public { abi.encode(a); }
}
```
also has it - but I'd also have called that a "nested array". I can test that explicitly as well, though.
I can also add tests for one-dimensional arrays, but they don't have the bug.

Reply:
ok, would be good to have that as a test case.

And yeah, I'd still add cases for 1D arrays just for better coverage. It also makes it easy to verify during review that this case is really bug-free.